x/distribution/spec/04_messages.md

```diff
@@ -27,163 +30,91 @@ func (k Keeper) SetWithdrawAddr(ctx sdk.Context, delegatorAddr sdk.AccAddress, w
 
 ## MsgWithdrawDelegatorReward
 
-under special circumstances a delegator may wish to withdraw rewards from only
-a single validator. 
-
-+++ https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/cosmos/distribution/v1beta1/tx.proto#L42-L50
-
-```go
-func WithdrawDelegationReward(delegatorAddr, validatorAddr, withdrawAddr sdk.AccAddress) 
-    height = GetHeight()
-    
-    // get all distribution scenarios
-    pool = staking.GetPool() 
-    feePool = GetFeePool() 
-    delInfo = GetDelegationDistInfo(delegatorAddr,
-                    validatorAddr)
-    valInfo = GetValidatorDistInfo(validatorAddr)
-    validator = GetValidator(validatorAddr)
-
-    feePool, withdraw = delInfo.WithdrawRewards(feePool, valInfo, height, pool.BondedTokens, 
-               validator.Tokens, validator.DelegatorShares, validator.Commission)
-
-    SetFeePool(feePool) 
-    SendCoins(distributionModuleAcc, withdrawAddr, withdraw.TruncateDecimal())
-```
-
-
-### Withdraw Validator Rewards All
-
-When a validator wishes to withdraw their rewards it must send
-array of `MsgWithdrawDelegatorReward`. Note that parts of this transaction logic are also
-triggered each with any change in individual delegations, such as an unbond,
-redelegation, or delegation of additional tokens to a specific validator. This
-transaction withdraws the validators commission fee, as well as any rewards
-earning on their self-delegation.
-
-```go
-
-func WithdrawValidatorRewardsAll(operatorAddr, withdrawAddr sdk.AccAddress)
-
-    height = GetHeight()
-    feePool = GetFeePool() 
-    pool = GetPool() 
-    ValInfo = GetValidatorDistInfo(delegation.ValidatorAddr)
-    validator = GetValidator(delegation.ValidatorAddr)
+A delegator can withdraw its rewards.
+Internally in the distribution module, this is treated exactly as if the delegator simply started a new delegation of the same value, simultaneously removing the previous delegation with associated rewards.
```

Comment:
```suggestion
Internally in the distribution module, this transaction simultaneously removes the previous delegation with associated rewards, the same as if the delegator simply started a new delegation of the same value.
```

Reply:
somehow putting what the transaction does makes more sense that saying what it's like